app/views/fines/_fines_table.html.erb

```diff
@@ -15,7 +15,7 @@
   <% @fines.each do |fine| %>
     <tr>
       <td>
-        <%= render BibItemComponent.new bibitem: fine %>
+        <%= (render BibItemComponent.new(bibitem: fine) unless fine.catkey.nil?) || fine.title %>
```

Comment:
I wonder if it'd be better to remove the `render?` method from `BibItemComponent` and instead build this logic of just printing the title by itself if there is no catkey (from the component itself)?

Reply:
Two things: We have a `render?` method in the component that validates on catkey so if there is no catkey it does not show which is the case for bills like misc charges. Also this is only for bills afaik (maybe it is not, I should check with Maryam on this) then we will need to check for that as well. I thought about changing the component too but kinda ended up with thinking not to complicate the component further since it is used by checkouts, holds and fines. But yes it can be done like that as well, which would need some logic changes and spec updates.

Reply:
Yeah it's just a little awkward as is to my eyes. Like `catkey` is getting checked for `nil` twice. I get the problem, like rendering an empty BibItemComponent is truthy.

Reply:
There is no item(bib) present in this case so it kinda made sense not to use the BibItemComponent. 
But if this custom title is also for checkouts and holds as well then it is one less thing to check in the component so I think we can think about changing the `render?` logic there.

Reply:
I see, yes that makes sense. It's an awkward problem for sure.

Reply:
It is an odd edge case for bills only. Maryam confirmed that it is only for bills, I am inclined toward not much changing the component but what are your thoughts?

Reply:
I wish rendering an empty view component would produce a nil so as to avoid having to double check that catkey is nil. What do you think of a helper to contain all this like 


```rb
def fine_title_display(fine: fine)
  render BibItemComponent.new(bibitem: fine) unless fine.catkey.nil?
  fine.title
end
```

I don't know I don't like this a lot more, but I think it is always better when there is less logic in the partial. 🤷 thoughts? If you feel good about this I'll approve/merge through.

Reply:
yeah adding a helper is a good idea, will do it.